statsmodels/miscmodels/betareg.py

```diff
@@ -0,0 +1,465 @@
+# -*- coding: utf-8 -*-
+
+u"""
+Beta regression for modeling rates and proportions.
+
+References
+----------
+Grün, Bettina, Ioannis Kosmidis, and Achim Zeileis. Extended beta regression
+in R: Shaken, stirred, mixed, and partitioned. No. 2011-22. Working Papers in
+Economics and Statistics, 2011.
+
+Smithson, Michael, and Jay Verkuilen. "A better lemon squeezer?
+Maximum-likelihood regression with beta-distributed dependent variables."
+Psychological methods 11.1 (2006): 54.
+"""
+from __future__ import print_function
+import numpy as np
+import pandas as pd
+import statsmodels.api as sm
+from statsmodels.tools.decorators import cache_readonly
+import patsy
+
+from scipy.special import gammaln as lgamma
+
+from statsmodels.base.model import (GenericLikelihoodModel,
+                                    GenericLikelihoodModelResults)
+from statsmodels.genmod.families import Binomial
+
+Logit = sm.families.links.logit
+
+_init_example = """
+
+    Beta regression with default of logit-link for exog and log-link
+    for precision.
+
+    >>> mod = Beta(endog, exog)
+    >>> rslt = mod.fit()
+    >>> print(rslt.summary())
+
+    We can also specify a formula and a specific structure and use the
+    identity-link for precision.
+
+    >>> from sm.families.links import identity
+    >>> Z = patsy.dmatrix('~ temp', dat, return_type='dataframe')
+    >>> mod = Beta.from_formula('iyield ~ C(batch, Treatment(10)) + temp',
+    ...                         dat, exog_precision=Z, link_precision=identity())
```

Comment:
add example when exog_precision is defined by formula

usage for it is not clear, from_formula uses `exog_precision_formula` keyword